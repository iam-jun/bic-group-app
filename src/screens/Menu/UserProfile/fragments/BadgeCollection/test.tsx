import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
