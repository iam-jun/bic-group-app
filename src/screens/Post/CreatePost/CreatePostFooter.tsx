import React, {FC} from 'react';
import {StyleSheet, View} from 'react-native';
import {useTheme} from 'react-native-paper';

import {ITheme} from '~/theme/interfaces';

import PostToolbar from '~/screens/Post/components/PostToolbar';
import MentionBar from '~/beinComponents/inputs/MentionInput/MentionBar';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

export interface CreatePostFooterProps {
  toolbarRef?: any;
  loading?: boolean;
  onPressBack?: () => void;
  imageDisabled?: boolean;
  videoDisabled?: boolean;
  fileDisabled?: boolean;
}

const CreatePostFooter: FC<CreatePostFooterProps> = ({
  toolbarRef,
  loading,
  onPressBack,
  imageDisabled,
  videoDisabled,
  fileDisabled,
}: CreatePostFooterProps) => {
  const showMentionValue = useSharedValue(0);

  const theme = useTheme() as ITheme;
  const styles = createStyle(theme);

  const mentionContainerStyle = useAnimatedStyle(() => ({
    position: 'absolute',
    right: 0,
    left: 0,
    opacity: showMentionValue.value,
  }));

  const onVisibleMentionBar = (isVisible: boolean) => {
    if (isVisible) {
      showMentionValue.value = withTiming(1);
    } else {
      showMentionValue.value = withTiming(0);
    }
  };

  return (
    <View>
      <PostToolbar
        toolbarRef={toolbarRef}
        disabled={loading}
        onPressBack={onPressBack}
        imageDisabled={imageDisabled}
        videoDisabled={videoDisabled}
        fileDisabled={fileDisabled}
      />
      <Animated.View
        testID={'create_post_footer.mention_bar_container'}
        style={mentionContainerStyle}>
        <MentionBar
          testID={'mention_bar'}
          onVisible={onVisibleMentionBar}
          style={styles.mentionBar}
        />
      </Animated.View>
    </View>
  );
};

const createStyle = (theme: ITheme) => {
  const {colors} = theme;
  return StyleSheet.create({
    mentionBar: {
      borderColor: colors.borderDivider,
    },
  });
};

export default CreatePostFooter;
