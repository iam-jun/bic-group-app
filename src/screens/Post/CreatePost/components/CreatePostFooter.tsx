import React, {FC} from 'react';
import {StyleSheet, View} from 'react-native';
import {ExtendedTheme, useTheme} from '@react-navigation/native';

import PostToolbar from '~/screens/Post/components/PostToolbar';
import MentionBar from '~/beinComponents/inputs/MentionInput/MentionBar';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

export interface CreatePostFooterProps {
  toolbarRef?: any;
  loading?: boolean;
  imageDisabled?: boolean;
  videoDisabled?: boolean;
  fileDisabled?: boolean;
  onPressBack?: () => void;
}

const CreatePostFooter: FC<CreatePostFooterProps> = ({
  toolbarRef,
  loading,
  imageDisabled,
  videoDisabled,
  fileDisabled,
  onPressBack,
}: CreatePostFooterProps) => {
  const showMentionValue = useSharedValue(0);

  const theme: ExtendedTheme = useTheme();
  const styles = createStyle(theme);

  const mentionContainerStyle = useAnimatedStyle(() => ({
    position: 'absolute',
    right: 0,
    left: 0,
    opacity: showMentionValue.value,
  }));

  const onVisibleMentionBar = (isVisible: boolean) => {
    if (isVisible) {
      showMentionValue.value = withTiming(1);
    } else {
      showMentionValue.value = withTiming(0);
    }
  };

  return (
    <View>
      <PostToolbar
        toolbarRef={toolbarRef}
        disabled={loading}
        imageDisabled={imageDisabled}
        videoDisabled={videoDisabled}
        fileDisabled={fileDisabled}
        onPressBack={onPressBack}
      />
      <Animated.View
        testID={'create_post_footer.mention_bar_container'}
        style={mentionContainerStyle}>
        <MentionBar
          testID={'mention_bar'}
          onVisible={onVisibleMentionBar}
          style={styles.mentionBar}
        />
      </Animated.View>
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const {colors} = theme;
  return StyleSheet.create({
    mentionBar: {
      borderColor: colors.neutral5,
    },
  });
};

export default CreatePostFooter;
