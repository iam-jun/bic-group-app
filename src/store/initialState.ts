import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
