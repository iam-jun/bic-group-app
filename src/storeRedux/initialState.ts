import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
