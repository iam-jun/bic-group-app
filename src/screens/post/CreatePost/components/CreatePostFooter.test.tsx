import React from 'react';

import { configureStore, renderWithRedux } from '~/test/testUtils';
import CreatePostFooter from '~/screens/post/CreatePost/components/CreatePostFooter';
import initialState from '~/storeRedux/initialState';
import useMentionInputStore from '~/beinComponents/inputs/MentionInput/store';
import { MENTION_USER } from '~/test/mock_data/mention';

describe('CreatePostFooter component', () => {
  const mockStore = configureStore([]);
  const onPressSetting = jest.fn();
  const onPressTags = jest.fn();
  const onPressSeries = jest.fn();

  it('renders correctly mention bar with items', async () => {
    useMentionInputStore.setState((state) => {
      state.data = [MENTION_USER];
      return state;
    });
    const storeData = { ...initialState };
    const store = mockStore(storeData);

    const wrapper = renderWithRedux(<CreatePostFooter
      onPressSetting={onPressSetting}
      onPressTags={onPressTags}
      onPressSeries={onPressSeries}
    />, store);
    const items = wrapper.getAllByTestId('mention_bar.item');
    expect(items?.length).toBe(1);
  });
});
