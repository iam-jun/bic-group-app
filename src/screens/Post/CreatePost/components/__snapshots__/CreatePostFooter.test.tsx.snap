// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePostFooter component renders correctly 1`] = `
<View>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E3E6F2",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 52,
          "paddingHorizontal": 16,
        }
      }
      testID="post_toolbar"
    >
      <View
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="post_toolbar.markdown_preview"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="post_toolbar.add_photo"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="post_toolbar.add_video"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="post_toolbar.add_file"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "height": 0,
          "width": "100%",
        }
      }
    />
  </View>
  <View>
    <View>
      <View
        style={
          Object {
            "height": 1067.2,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#181C32",
                  "fontFamily": "BeVietnamPro-SemiBold",
                  "fontSize": 15,
                },
                Object {},
                undefined,
              ]
            }
          >
            Review Markdown
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#7335C0",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginRight": 4,
                "padding": 8,
              }
            }
            testID="header.button"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
              testID="button_wrapper.content"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#181C32",
                      "fontFamily": "BeVietnamPro-Medium",
                      "fontSize": 15,
                    },
                    Object {
                      "color": "#7335C0",
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Done
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#F8F9FF",
              "flex": 1,
              "marginHorizontal": 12,
              "marginTop": 16,
              "paddingHorizontal": 8,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 4,
                  "height": 48,
                  "marginRight": 8,
                  "width": 48,
                }
              }
              testID="avatar"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#A4A9B7",
                    "borderRadius": 24,
                    "height": 48,
                    "width": 48,
                  }
                }
                testID="avatar_container"
              >
                <View
                  onLayout={[Function]}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "position": "relative",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#f9f9f9",
                          "borderRadius": 24,
                          "height": 48,
                          "justifyContent": "center",
                          "width": 48,
                        }
                      }
                      testID="Image__placeholder"
                    />
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 24,
                          "height": 48,
                          "width": 48,
                        },
                      ]
                    }
                  >
                    <FastImageView
                      onFastImageError={[Function]}
                      onFastImageLoadEnd={[Function]}
                      resizeMode="cover"
                      source={null}
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                      testID="avatar.image"
                    />
                  </View>
                </View>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#181C32",
                      "fontFamily": "BeVietnamPro-SemiBold",
                      "fontSize": 13,
                    },
                    Object {},
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#181C32",
                        "fontFamily": "BeVietnamPro-Medium",
                        "fontSize": 13,
                      },
                      Object {
                        "color": "#767D93",
                      },
                      Object {
                        "marginRight": 4,
                      },
                    ]
                  }
                >
                  to
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#181C32",
                        "fontFamily": "BeVietnamPro-SemiBold",
                        "fontSize": 13,
                      },
                      Object {},
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <RCTScrollView
            style={
              Object {
                "marginTop": 24,
              }
            }
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
      }
    }
    testID="create_post_footer.mention_bar_container"
  />
</View>
`;
